agr_java_core/src/main/java/org/alliancegenome/es/util/IndexManager.java

```diff
@@ -293,16 +309,22 @@ public void createSnapShot(String repo, String snapShotName, List<String> indice
         try {
             log.info("Creating Snapshot: " + snapShotName + " in: " + repo + " with: " + indices);
 
+            requestFinished = false;
+
             CreateSnapshotRequest request = new CreateSnapshotRequest();
             request.repository(repo);
             request.snapshot(snapShotName);
             request.indices(indices);
             request.waitForCompletion(true);
 
-            getClient().snapshot().create(request, RequestOptions.DEFAULT);
+            getClient().snapshot().createAsync(request, RequestOptions.DEFAULT, requestListener);
+            
+            while(!requestFinished) {
+                TimeUnit.SECONDS.sleep(5);
+            }
 
```

Comment:
The goal here is that this method HAS to wait until it's finished. However `getClient().snapshot().create(request, RequestOptions.DEFAULT);` times out at 1800 seconds.